pkg/providers/aws/provider_redis.go

```diff
@@ -501,7 +501,7 @@ func (p *RedisProvider) DeleteRedis(ctx context.Context, r *v1alpha1.Redis) (cro
 	}
 
 	// network manager required for cleaning up network.
-	networkManager := NewNetworkManager(sess, p.Client, logger)
+	networkManager := NewNetworkManager(sess, p.Client, logger, isSTSCluster(ctx, p.Client))
```

Comment:
@laurafitzgerald That function was created previously and has been running without issues for everyone using the STS feature branch so far. If the function `isSTSCluster` gets invoked from a non STS cluster it will return false. I couldn't identify any possible errors down the line as all error cases seem to be handled.